Lib/_collections_abc.py

```diff
@@ -106,11 +106,11 @@ async def _ag(): yield
 ### ONE-TRICK PONIES ###
 
 def _check_methods(C, *methods):
-    mro = C.__mro__
+    mro_dicts = [B.__dict__ for B in C.__mro__]
     for method in methods:
-        for B in mro:
-            if method in B.__dict__:
-                if B.__dict__[method] is None:
+        for base_dict in mro_dicts:
+            if method in base_dict:
+                if base_dict[method] is None:
                     return NotImplemented
                 break
```

Comment:
```suggestion
        for base_dict in mro_dicts:
            method_impl = base_dict.get(method, sentinel)
            if method_impl is None:
                return NotImplemented
            if method_impl is not sentinel:
                break
```
This avoids calculating hash and looking for key in dict twice. sentinel object can be created at the top of the method.

Reply:
That looks like a clear win, good eye! Might make sense to do the check optimistically, ie: consider the sentinel more likely than the None

```python
if method_impl is sentinel:
    continue
if method_impl is None:
    return NotImplemented
break
```